mm2src/coins/sia.rs

```diff
@@ -0,0 +1,564 @@
+use super::{CoinBalance, HistorySyncState, MarketCoinOps, MmCoin, RawTransactionFut, RawTransactionRequest, SwapOps,
+            TradeFee, TransactionEnum, TransactionFut};
+use crate::{coin_errors::MyAddressError, BalanceFut, CanRefundHtlc, CheckIfMyPaymentSentArgs, CoinFutSpawner,
+            ConfirmPaymentInput, FeeApproxStage, FoundSwapTxSpend, MakerSwapTakerCoin, MmCoinEnum,
+            NegotiateSwapContractAddrErr, PaymentInstructionArgs, PaymentInstructions, PaymentInstructionsErr,
+            PrivKeyBuildPolicy, RawTransactionResult, RefundPaymentArgs, RefundResult, SearchForSwapTxSpendInput,
+            SendMakerPaymentSpendPreimageInput, SendPaymentArgs, SignRawTransactionRequest, SignatureResult,
+            SpendPaymentArgs, TakerSwapMakerCoin, TradePreimageFut, TradePreimageResult, TradePreimageValue,
+            TransactionResult, TxMarshalingErr, UnexpectedDerivationMethod, ValidateAddressResult, ValidateFeeArgs,
+            ValidateInstructionsErr, ValidateOtherPubKeyErr, ValidatePaymentError, ValidatePaymentFut,
+            ValidatePaymentInput, ValidatePaymentResult, VerificationResult, WaitForHTLCTxSpendArgs, WatcherOps,
+            WatcherReward, WatcherRewardError, WatcherSearchForSwapTxSpendInput, WatcherValidatePaymentInput,
+            WatcherValidateTakerFeeInput, WithdrawFut, WithdrawRequest};
+use crate::{DexFee, ValidateWatcherSpendInput};
+use async_trait::async_trait;
+use common::executor::AbortedError;
+use futures::{FutureExt, TryFutureExt};
+use futures01::Future;
+use keys::KeyPair;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use mm2_number::{BigDecimal, MmNumber};
+use rpc::v1::types::Bytes as BytesJson;
+use serde_json::Value as Json;
+use std::ops::Deref;
+use std::sync::Arc;
+
+pub mod http_client;
+use http_client::{SiaApiClient, SiaApiClientError};
+
+use url::Url;
+
+use mm2_core::mm_ctx::MmWeak;
+
+#[derive(Clone)]
+pub struct SiaCoin(SiaArc);
+#[derive(Clone)]
+pub struct SiaArc(Arc<SiaCoinFields>);
+
+#[derive(Debug, Display)]
+pub enum SiaConfError {
+    #[display(fmt = "'foo' field is not found in config")]
+    Foo,
+    Bar(String),
+}
+
+pub type SiaConfResult<T> = Result<T, MmError<SiaConfError>>;
+
+#[derive(Debug)]
+pub struct SiaCoinConf {
+    ticker: String,
+    pub foo: u32,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SiaCoinActivationParams {
```

Comment:
Things you will need to add to these activation params are
https://github.com/KomodoPlatform/komodo-defi-framework/blob/af571608c34f38d82e3c421f9bd61c5265db0cf8/mm2src/coins/utxo.rs#L1450-L1451
https://github.com/KomodoPlatform/komodo-defi-framework/blob/af571608c34f38d82e3c421f9bd61c5265db0cf8/mm2src/coins/utxo.rs#L1457-L1460
But both will be needed when you start HD wallet implementation